tests/smoke_tests/smoke_tests_utils.py

```diff
@@ -296,25 +319,39 @@ def run_one_test(test: Test) -> None:
         )
         try:
             proc.wait(timeout=test.timeout)
+        except subprocess.TimeoutExpired as e:
+            # Kill the current process.
+            proc.terminate()
+            # Raise to interrupt retry loop.
+            raise e
+        return proc.returncode
+
+    for command in test.commands:
+        try:
+            return_code = 0
+            if isinstance(command, RetriableCommand):
+                for i in range(command.max_attempts):
+                    return_code = run_command(command.command, i)
+                    if return_code == 0:
+                        break
+                    time.sleep(command.retry_interval)
+            else:
+                return_code = run_command(command)
         except subprocess.TimeoutExpired as e:
             flush()
             test.echo(f'Timeout after {test.timeout} seconds.')
             test.echo(str(e))
             write(f'Timeout after {test.timeout} seconds.\n')
             flush()
-            # Kill the current process.
-            proc.terminate()
-            proc.returncode = 1  # None if we don't set it.
-            break
-
-        if proc.returncode:
+            return_code = 1
+        if return_code:
             break
 
     style = colorama.Style
     fore = colorama.Fore
-    outcome = (f'{fore.RED}Failed{style.RESET_ALL} (returned {proc.returncode})'
-               if proc.returncode else f'{fore.GREEN}Passed{style.RESET_ALL}')
-    reason = f'\nReason: {command}' if proc.returncode else ''
+    outcome = (f'{fore.RED}Failed{style.RESET_ALL} (returned {return_code})'
```

Comment:
minor: we use `returncode` in everywhere else. Let's keep them consistent?
```suggestion
    outcome = (f'{fore.RED}Failed{style.RESET_ALL} (returned {returncode})'
```

Reply:
good catch!